packages/pubsub/src/index.js

```diff
@@ -79,43 +67,151 @@ function PubSub(options) {
     return new PubSub(options);
   }
 
-  this.defaultBaseUrl_ = 'pubsub.googleapis.com';
-  this.determineBaseUrl_(options.apiEndpoint);
-
-  var config = {
-    baseUrl: this.baseUrl_,
-    customEndpoint: this.customEndpoint_,
-    protosDir: path.resolve(__dirname, '../protos'),
-    protoServices: {
-      Publisher: {
-        path: 'google/pubsub/v1/pubsub.proto',
-        service: 'pubsub.v1'
-      },
-      Subscriber: {
-        path: 'google/pubsub/v1/pubsub.proto',
-        service: 'pubsub.v1'
-      }
-    },
-    scopes: [
-      'https://www.googleapis.com/auth/pubsub',
-      'https://www.googleapis.com/auth/cloud-platform'
-    ],
-    packageJson: require('../package.json')
-  };
+  this.options = extend({
+    scopes: v1.ALL_SCOPES
```

Comment:
We will need to add:

```js
libName: 'gccl',
libVersion: PKG.version
```

where PKG is required earlier to be ../package.json.